packages/@react-aria/ssr/src/SSRProvider.tsx

```diff
@@ -71,7 +69,9 @@ function LegacySSRProvider(props: SSRProviderProps): JSX.Element {
 
   return (
     <SSRContext.Provider value={value}>
-      {props.children}
+      <IsSSRContext.Provider value={isSSR}>
```

Comment:
Apparently SSR was kinda broken in React 16/17. The problem was that the SSR context would reset whenever `isSSR` changed, and the children would not necessarily re-render in the same order, which could result in duplicate ids. The fix is to split that into two contexts so the ids don't change value.